src/coreclr/vm/jitinterface.cpp

```diff
@@ -10859,41 +10846,68 @@ void CEECodeGenInfo::getHelperFtn(CorInfoHelpFunc    ftnNum,               /* IN
                     {
                         // Cache it for future uses to avoid taking the lock again.
                         hlpFinalTierAddrTable[dynamicFtnNum] = finalTierAddr;
-                        if (pNativeEntrypoint != NULL)
-                        {
-                            pNativeEntrypoint->accessType = IAT_VALUE;
-                            pNativeEntrypoint->addr = finalTierAddr;
-                        }
+                        accessType = IAT_VALUE;
+                        targetAddr = finalTierAddr;
                         goto exit;
                     }
                 }
             }
 
             if (IndirectionAllowedForJitHelper(ftnNum))
             {
+#ifdef FEATURE_PORTABLE_ENTRYPOINTS
+                accessType = IAT_VALUE;
+                targetAddr = (LPVOID)helperMD->GetPortableEntryPoint();
+#else // !FEATURE_PORTABLE_ENTRYPOINTS
                 Precode* pPrecode = helperMD->GetPrecode();
                 _ASSERTE(pPrecode->GetType() == PRECODE_FIXUP);
-                if (pNativeEntrypoint != NULL)
-                {
-                    pNativeEntrypoint->accessType = IAT_PVALUE;
-                    pNativeEntrypoint->addr = ((FixupPrecode*)pPrecode)->GetTargetSlot();
-                }
+                accessType = IAT_PVALUE;
+                targetAddr = ((FixupPrecode*)pPrecode)->GetTargetSlot();
+#endif // FEATURE_PORTABLE_ENTRYPOINTS
                 goto exit;
             }
         }
 
-        pfnHelper = LoadDynamicJitHelper((DynamicCorInfoHelpFunc)dynamicFtnNum).pfnHelper;
+        pfnHelper = LoadDynamicJitHelper((DynamicCorInfoHelpFunc)dynamicFtnNum);
     }
 
-    _ASSERTE(pfnHelper != NULL);
+    //
+    // Static helpers
+    //
 
-    if (pNativeEntrypoint != NULL)
+    _ASSERTE(pfnHelper != (TADDR)NULL);
+
+    accessType = IAT_VALUE;
+
+#ifdef FEATURE_PORTABLE_ENTRYPOINTS
+    targetAddr = (LPVOID)hlpFuncEntryPoints[ftnNum];
+    if (targetAddr == NULL)
+#endif // FEATURE_PORTABLE_ENTRYPOINTS
     {
-        pNativeEntrypoint->accessType = IAT_VALUE;
-        pNativeEntrypoint->addr = (LPVOID)GetEEFuncEntryPoint(pfnHelper);
+        TADDR entryPoint = GetEEFuncEntryPoint(pfnHelper);
```

Comment:
```suggestion
        TADDR entryPoint = pfnHelper;
```
`GetEEFuncEntryPoint` macro does make sense here. It used to extract actual code on platforms where function pointers did not point to actual code (we do not have any platform like that currently , and it is also used to DACize pointers to runtime functions (this file is not DAcized so it does not apply here).